powerauth-backend-tests/src/test/java/com/wultra/security/powerauth/test/v31/PowerAuthActivationTest.java

```diff
@@ -0,0 +1,507 @@
+        model.setResultStatusObject(resultStatusObject);
+        ObjectStepLogger stepLoggerPrepare = new ObjectStepLogger(System.out);
+        new PrepareActivationStep().execute(stepLoggerPrepare, model.toMap());
+        assertTrue(stepLoggerPrepare.getResult().isSuccess());
+        assertEquals(200, stepLoggerPrepare.getResponse().getStatusCode());
+
+        EciesEncryptedResponse eciesResponse = (EciesEncryptedResponse) stepLoggerPrepare.getResponse().getResponseObject();
+        assertNotNull(eciesResponse.getEncryptedData());
+        assertNotNull(eciesResponse.getMac());
+
+        // Verify activation status
+        GetStatusStepModel statusModel = new GetStatusStepModel();
+        statusModel.setResultStatusObject(resultStatusObject);
+        statusModel.setHeaders(new HashMap<>());
+        statusModel.setUriString(config.getPowerAuthIntegrationUrl());
+        statusModel.setVersion("3.1");
+
+        ObjectStepLogger stepLoggerStatus = new ObjectStepLogger(System.out);
+        new GetStatusStep().execute(stepLoggerStatus, statusModel.toMap());
+        assertTrue(stepLoggerStatus.getResult().isSuccess());
+        assertEquals(200, stepLoggerStatus.getResponse().getStatusCode());
+        ActivationStatusRequest request = (ActivationStatusRequest) stepLoggerStatus.getRequest().getRequestObject();
+        assertNotNull(request.getChallenge());
+        ObjectResponse<ActivationStatusResponse> responseObject = (ObjectResponse<ActivationStatusResponse>) stepLoggerStatus.getResponse().getResponseObject();
+        ActivationStatusResponse response = responseObject.getResponseObject();
+        assertEquals(initResponse.getActivationId(), response.getActivationId());
+        assertNotNull(response.getNonce());
+        assertNull(response.getCustomObject());
+
+        // Get transport key
+        String transportMasterKeyBase64 = (String) model.getResultStatusObject().get("transportMasterKey");
+        SecretKey transportMasterKey = config.getKeyConversion().convertBytesToSharedSecretKey(BaseEncoding.base64().decode(transportMasterKeyBase64));
+
+        // Verify activation status blob
+        byte[] challengeData = BaseEncoding.base64().decode(request.getChallenge());
+        byte[] nonceData = BaseEncoding.base64().decode(response.getNonce());
+        byte[] cStatusBlob = BaseEncoding.base64().decode(response.getEncryptedStatusBlob());
+        ActivationStatusBlobInfo statusBlob = activation.getStatusFromEncryptedBlob(cStatusBlob, challengeData, nonceData, transportMasterKey);
+        assertTrue(statusBlob.isValid());
+        assertEquals(0x2, statusBlob.getActivationStatus());
+        assertEquals(10, statusBlob.getMaxFailedAttempts());
+        assertEquals(0, statusBlob.getFailedAttempts());
+        assertEquals(3, statusBlob.getCurrentVersion());
+        assertEquals(3, statusBlob.getUpgradeVersion());
+        assertArrayEquals(CounterUtil.getCtrData(model, stepLoggerStatus), statusBlob.getCtrData());
```

Comment:
We should probably verify the counter lookahead window in this test method for version `3.1`.

Reply:
OK